ParseLoginUI/src/main/java/com/parse/ParseQueryRecyclerViewAdapter.java

```diff
@@ -0,0 +1,611 @@
+package com.parse;
```

Comment:
We could just use the internal implementation of `ParseObject.getClassName(Class)`, but I'm hesitant to publicize `Parse.isLocalDatastoreEnabled()` since we might move or rename that...

Could we use a try/catch the `IllegalStateException` from `getCachePolicy` instead?